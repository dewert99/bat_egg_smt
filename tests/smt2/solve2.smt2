(declare-sort S1)
(declare-fun a () S1)
(declare-fun b () S1)
(declare-fun x () Bool)
(assert (or (= a b) (= x (not x))))
(declare-fun f (S1) Bool)
(check-sat-assuming ( (and (f a) (not (f b)))))
(get-unsat-core)
(check-sat-assuming ((f a)))
(check-sat-assuming ((not (f a))))