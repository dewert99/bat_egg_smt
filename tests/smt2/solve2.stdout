unsat
((and (f a) (not (f b))))
sat
((a S1!val!2)
 (b S1!val!2)
 (x true)
 ((f a) true)
 ((f b) true))
(
 (define-fun a () S1 S1!val!2)
 (define-fun b () S1 S1!val!2)
 (define-fun f ((x!0 S1)) Bool
  (ite (= x!0 S1!val!2) true
   f!default))
 (define-fun x () Bool true)
)
sat
(
 (define-fun a () S1 S1!val!2)
 (define-fun b () S1 S1!val!2)
 (define-fun f ((x!0 S1)) Bool
  (ite (= x!0 S1!val!2) false
   f!default))
 (define-fun x () Bool true)
)
((a S1!val!2)
 (b S1!val!2)
 (x true)
 ((f a) false)
 ((f b) false))
